Margetest 6.16 rc2 0619 3

myangsynapticscom opened the pull request.

myangsynaptics committed:
Input: synaptics-rmi4- Add a new feature for Forcepad.

Forcepad devices will use F21, for click simulation
due to lack of a metal button, so we add F21 support
to make forcepad support click function.

Signed-off-by: Marge Yang <Marge.Yang@tw.synaptics.com>

myangsynaptics committed:
Input: synaptics-rmi4- Add a new feature for Forcepad.

Forcepad devices will use F21, for click simulation
due to lack of a metal button, so we add F21 support
to make forcepad support click function.

Signed-off-by: Marge Yang <Marge.Yang@tw.synaptics.com>

myangsynaptics committed:
Input: synaptics-rmi4- Add a new feature for Forcepad.

Forcepad devices will use F21, for click simulation
due to lack of a metal button, so we add F21 support
to make forcepad support click function.

Signed-off-by: Marge Yang <Marge.Yang@tw.synaptics.com>

vhuag commented on drivers/input/rmi4/rmi_f21.c:
```diff
@@ -32,14 +32,14 @@ static irqreturn_t rmi_f21_attention(int irq, void *ctx)
 			return IRQ_HANDLED;
 		}
-		bpressed = (((u8 *)drvdata->attn_data.data)[f21->max_number_Of_finger] &
+		pressed = (((u8 *)drvdata->attn_data.data)[f21->max_number_Of_finger] &
```
We should use attn data size -1 here right?  otherwise we'll get wrong index in old forcepad.

vhuag commented on drivers/input/rmi4/rmi_f21.c:
```diff
@@ -51,11 +51,11 @@ static irqreturn_t rmi_f21_attention(int irq, void *ctx)
 			return IRQ_RETVAL(error);
 		}
-		bpressed = (f21->data_regs[f21->sensor_count * 2] &
+		pressed = (f21->data_regs[f21->sensor_count * 2] &
```
Same here, will we get wrong index if this is new forcepad?

myangsynaptics committed:
Input: synaptics-rmi4- Add a new feature for Forcepad.

Forcepad devices will use F21, for click simulation
due to lack of a metal button, so we add F21 support
to make forcepad support click function.

Signed-off-by: Marge Yang <Marge.Yang@tw.synaptics.com>
